CMakeLists.txt

```diff
@@ -21,22 +24,17 @@ if(NOT TARGET json-hpp)
     endif()
 
     # create an interface-library for simple cmake-linking
+    include_directories(${NLOHMANN_JSON_REALPATH})
```

Comment:
Using `include_directories()` is discouraged since some time. Only in very rare cases this should be used. Did you remove the `target_xxx()`-stuff because it does not allow you to create the package-files with export?

Reply:
I believe it caused problems on the import. Since writing this and trying to incorporate these cmake build scripts in other projects I *think* I now understand it to be the case that if your cmake project relies on other projects (that you include via find_package, say) any project that includes yours must also include the dependencies explicitly, which was sort of non-intuitive. I'm not sure how to get around that though.

Reply:
I think [`INTERFACE_LINK_LIBRARIES`](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_LINK_LIBRARIES.html) allows you to implicitly import dependencies.
So, something like (untested!)
set_property(TARGET json-schema-validator PROPERTY INTERFACE_LINK_LIBRARIES json-hpp)

Imo @pboettch is right and using `include_directories` instead of `target_include_directories` is very bad style and could lead to more problems for larger projects and therefore shouldn't be merged like this.